core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -577,16 +577,18 @@ public static String substring(String c, int s) {
   /** SQL SUBSTRING(string FROM ... FOR ...) function. */
   public static String substring(String c, int s, int l) {
     int lc = c.length();
-    int e = s + l;
+    long e = (long) s + (long) l;
```

Comment:
Should we adjust the other `substring()` signatures in this class as well for consistency?